oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/core/expression/OAuth2Expressions.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.expression;
+
+import org.springframework.security.core.Authentication;
+
+/**
+ * A contract for the OAuth2 SpEL expressions entry point
+ *
+ * @author Josh Cummings
+ * @since 5.1
+ */
+public interface OAuth2Expressions {
+
+	/**
+	 * Retreive an OAuth2 token attribute from the provided {@link Authentication}
+	 *
+	 * @param authentication
+	 * @param name
+	 * @return
+	 */
+	Object attribute(Authentication authentication, String name);
```

Comment:
Maybe we should change this to `hasAttribute(Authentication authentication, String name, Object value)`?

Reply:
I'd avoid adding an equals operato since we can get the attribute and use SpEL to see if it is equal to a value

Reply:
I think that the one exception to what you are saying @rwinch may be `hasScope`. Otherwise, agreed that it is nice to keep logic in our expression methods light so that they can be more easily mixed and matched.